### PR TITLE
Add `target_compatible_with` on `copy_config` in `xz`

### PR DESCRIPTION
**Pull request by lalten**
Main change is to add `target_compatible_with` on `copy_config` as we ran into
```
ERROR: /home/user/.cache/bazel/_bazel_user/5f48ff965103b174f3c248651ebad24d/external/xz~/BUILD.bazel:71:10: configurable attribute "src" in @@xz~//:copy_config doesn't match this configuration. Would a default condition help?

Conditions checked:
 @@platforms//os:android
 @@xz~//:linux_arm64
 @@xz~//:linux_x86_64
 @@xz~//:osx_arm64
 @@xz~//:osx_x86_64
 @@platforms//os:ios
 @@platforms//os:watchos
 @@platforms//os:tvos
 @@platforms//os:windows

To see a condition's definition, run: bazel query --output=build <condition label>.

This instance of @@xz~//:copy_config has configuration identifier 5075a55. To inspect its configuration, run: bazel config 5075a55.

For more help, see https://bazel.build/docs/configurable-attributes#faq-select-choose-condition.
```
and the OS this is configured for isn't in that list.

The other change is to switch to using `overlay` instead of patches because it's almost impossible to read the patch files. The files are otherwise unchanged.

I would have loved to upgrade to https://github.com/tukaani-project/xz/releases/tag/v5.6.2 but I can only `./configure` for my local Linux and don't know how to produce `config.lzma-android.h` and friends.

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @Attempt3035, @Vertexwahn, modules you maintain (xz) have been updated in this PR. Please review the changes.

**Review by Attempt3035 — APPROVED** [review]
Looks good 👍

**Review by bazel-io — DISMISSED** [review]
Hello @bazelbuild/bcr-maintainers, all modules in this PR have been approved by their maintainers. Please take a final look to merge this PR.

File: modules/xz/5.4.5.bcr.4/overlay/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+    name = "copy_config",
+    src = selects.with_or({
+        "@platforms//os:android": "config.lzma-android.h",
+        ":linux_arm64": "config.lzma-linux-arm64.h",
+        ":linux_x86_64": "config.lzma-linux-x86_64.h",
+        ":osx_arm64": "config.lzma-osx-arm64.h",
+        ":osx_x86_64": "config.lzma-osx-x86_64.h",
+        ("@platforms//os:ios", "@platforms//os:watchos", "@platforms//os:tvos"): "apple_config",
+        "@platforms//os:windows": "config.lzma-windows.h",
+        "//conditions:default": [],
```
Comment (lalten) [inline modules/xz/5.4.5.bcr.4/overlay/BUILD.bazel:81]:
```suggestion
        "//conditions:default": ["config.lzma-linux-x86_64.h"],
```

**Comment by meteorcloudy** [comment]
Please fix the presubmit

**Review by Vertexwahn — APPROVED** [review]

**Review by bazel-io — APPROVED** [review]
Hello @bazelbuild/bcr-maintainers, all modules in this PR have been approved by their maintainers. Please take a final look to merge this PR.

**Review by Attempt3035 — APPROVED** [review]
Loving the overlay system btw! Makes jobs like this sooo much easier!

**Pull request merged**